tests/acceptance/test_streaming_data_measurement.py

```diff
@@ -0,0 +1,143 @@
+
+
+@pytest.mark.parametrize("num_responses", [1, 10, 100])
+def test___streaming_measurement_service___request_number_of_responses___receives_responses(
+    num_responses: int, stub_v2: v2_measurement_service_pb2_grpc.MeasurementServiceStub
+):
+    request = v2_measurement_service_pb2.MeasureRequest(
+        configuration_parameters=_get_configuration_parameters(num_responses=num_responses)
+    )
+
+    response_iterator = stub_v2.Measure(request)
+    responses = [response for response in response_iterator]
+    assert len(responses) == num_responses
+
+
+@pytest.mark.parametrize("data_size", [1, 10, 100])
+def test___streaming_measurement_service___request_data_cumulatively___receives_expected_amount_of_data(
+    data_size: int, stub_v2: v2_measurement_service_pb2_grpc.MeasurementServiceStub
+):
+    name = "testing-cumulative"
+    request = v2_measurement_service_pb2.MeasureRequest(
+        configuration_parameters=_get_configuration_parameters(
+            name=name,
+            data_size=data_size,
+            cumulative_data=True,
+        )
+    )
+
+    response_iterator = stub_v2.Measure(request)
+
+    expected_data: List[int] = []
+    index = 0
+    for response in response_iterator:
+        expected_data.extend(index for i in range(data_size))
+        expected = _get_serialized_measurement_outputs(name, index, expected_data)
+        assert expected == response.outputs.value
+        index += 1
+
+
+@pytest.mark.parametrize("data_size", [1, 10, 100])
+def test___streaming_measurement_service___specify_data_size___receives_expected_amount_of_data(
+    data_size: int, stub_v2: v2_measurement_service_pb2_grpc.MeasurementServiceStub
+):
+    name = "testing-not-cumulative"
+    request = v2_measurement_service_pb2.MeasureRequest(
+        configuration_parameters=_get_configuration_parameters(
+            name=name,
+            data_size=data_size,
+            cumulative_data=False,
+        )
+    )
+
+    response_iterator = stub_v2.Measure(request)
+
+    index = 0
+    for response in response_iterator:
+        expected_data = [index for i in range(data_size)]
+        expected = _get_serialized_measurement_outputs(name, index, expected_data)
+        assert expected == response.outputs.value
+        index += 1
+
+
+@pytest.mark.parametrize("error_on_index", [1, 5, 9])
+def test___streaming_measurement_service___specify_error_index___errors_at_expected_response(
+    error_on_index: int, stub_v2: v2_measurement_service_pb2_grpc.MeasurementServiceStub
+):
+    request = v2_measurement_service_pb2.MeasureRequest(
+        configuration_parameters=_get_configuration_parameters(error_on_index=error_on_index)
+    )
+
+    response_iterator = stub_v2.Measure(request)
+
+    with pytest.raises(grpc.RpcError, match=f"Errored at index {error_on_index}"):
+        for index, response in enumerate(response_iterator):
+            pass
+
+    assert (index + 1) == error_on_index
```

Comment:
3 sections: arrange, act, assert
```suggestion
    with pytest.raises(grpc.RpcError, match=f"Errored at index {error_on_index}"):
        for index, response in enumerate(response_iterator):
            pass
    assert (index + 1) == error_on_index
```